extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilder.scala

```diff
@@ -411,6 +469,26 @@ object PrivilegesBuilder {
         outputObjs += tablePrivileges(table, actionType = actionType)
         buildQuery(getQuery, inputObjs)
 
+      case "AppendData" => // DsV2 insert
+        val table = getPlanField[AnyRef]("table")
+        val tableIdent = getFieldVal[Option[Identifier]](table, "identifier")
+        outputObjs += tablePrivilegesWithIdentifier(tableIdent.get, actionType = INSERT)
+        buildQuery(getQuery, inputObjs)
+
+      case "UpdateTable" => // DsV2 update
+        val table = getPlanField[AnyRef]("table")
+        val tableIdent = getFieldVal[Option[Identifier]](table, "identifier")
+        outputObjs += tablePrivilegesWithIdentifier(tableIdent.get, actionType = UPDATE)
+      // todo INSERT ?
```

Comment:
if you add TODO, better to create an issue to track. otherwise, todo is never to be done.